src/components/localization/InstallationLanguage.jsx

```diff
@@ -103,89 +106,107 @@ class LanguageSelector extends React.Component {
                     }
                 }
             }
-            console.warn(`Locale with code ${localeCode} not found.`);
+            return null;
         };
 
-        // Returns a new instance of MenuItem from a given locale and with given prefix in it's key
-        // and id.
+        // Helper to create a menu item
         const createMenuItem = (locale, prefix) => {
             const isSelected = this.props.language === getLocaleId(locale);
 
-            // Creating a ref that will be applied to the selected language and cause it to scroll into view.
+            // Scroll into view if selected and not already scrolled
             const scrollRef = (isSelected && !foundSelected)
                 ? (ref) => {
                     if (ref) {
                         ref.scrollIntoView({ block: "center" });
+                        foundSelected = true;
                     }
                 }
                 : undefined;
 
-            const item = (
+            return (
                 <MenuItem
-                  id={idPrefix + "-" + prefix + getLocaleId(locale).split(".UTF-8")[0]}
-                  key={prefix + getLocaleId(locale)}
-                  isSelected={isSelected}
+                  id={`${idPrefix}-${prefix}-${getLocaleId(locale).split(".UTF-8")[0]}`}
+                  key={`${prefix}-${getLocaleId(locale)}`}
                   itemId={getLocaleId(locale)}
                   ref={scrollRef}
                   style={isSelected ? { backgroundColor: "var(--pf-v5-c-menu__list-item--hover--BackgroundColor)" } : undefined}
                 >
-                    {getLocaleNativeName(locale)}
+                    <div id="language-item">
+                        <div>
+                            <span>{getLocaleNativeName(locale)}</span>
+                            {isSelected && (
+                                <span id="selected-icon" aria-hidden="true">
+                                    <CheckIcon />
+                                </span>
+                            )}
+                        </div>
+                        <span id="item-translation">
+                            {getLanguageEnglishName(locale)}
```

Comment:
@garrett i think we need some clarification here.
What it the right column? IN the mockups it's inconsistent, sometimes translated locale to english, sometimes the native. Please clarify.

Also we would need some help to restrain the height of the menu, as it currently spans larger than the available space, and it creates the common issue with the double scrollbar.